src/biogeophys/BalanceCheckMod.F90

```diff
@@ -398,87 +572,178 @@ subroutine BalanceCheck( bounds, &
+               - qflx_drain_perched_grc(g)  &
+               - qflx_ice_runoff_grc(g)  &
+               - qflx_snwcp_discarded_liq_grc(g)  &
+               - qflx_snwcp_discarded_ice_grc(g)) * dtime
+       end do
+
+       errh2o_max_val = maxval(abs(errh2o_grc(bounds%begg:bounds%endg)))
+
+       if (errh2o_max_val > h2o_warning_thresh) then
+
+          indexg = maxloc( abs(errh2o_grc(bounds%begg:bounds%endg)), 1 ) + bounds%begg - 1
+          write(iulog,*)'WARNING:  grid cell-level water balance error ',&
+             ' nstep= ',nstep, &
+             ' local indexg= ',indexg,&
+             ' errh2o_grc= ',errh2o_grc(indexg)
+          if (errh2o_max_val > error_thresh .and. DAnstep > skip_steps .and. &
+              .not. get_for_testing_zero_dynbal_fluxes()) then
+
+             write(iulog,*)'CTSM model stopping because errh2o > ', error_thresh, ' mm'
+             write(iulog,*)'nstep                     = ',nstep
+             write(iulog,*)'errh2o_grc                = ',errh2o_grc(indexg)
+             write(iulog,*)'forc_rain                 = ',forc_rain_grc(indexg)*dtime
+             write(iulog,*)'forc_snow                 = ',forc_snow_grc(indexg)*dtime
+             write(iulog,*)'endwb_grc                 = ',endwb_grc(indexg)
+             write(iulog,*)'begwb_grc                 = ',begwb_grc(indexg)
+
+             write(iulog,*)'qflx_evap_tot             = ',qflx_evap_tot_grc(indexg)*dtime
+             write(iulog,*)'qflx_sfc_irrig            = ',qflx_sfc_irrig_grc(indexg)*dtime
+             write(iulog,*)'qflx_surf                 = ',qflx_surf_grc(indexg)*dtime
+             write(iulog,*)'qflx_qrgwl                = ',qflx_qrgwl_grc(indexg)*dtime
+             write(iulog,*)'qflx_drain                = ',qflx_drain_grc(indexg)*dtime
+             write(iulog,*)'qflx_ice_runoff           = ',qflx_ice_runoff_grc(indexg)*dtime
+             write(iulog,*)'qflx_snwcp_discarded_ice  = ',qflx_snwcp_discarded_ice_grc(indexg)*dtime
+             write(iulog,*)'qflx_snwcp_discarded_liq  = ',qflx_snwcp_discarded_liq_grc(indexg)*dtime
+             write(iulog,*)'deltawb                   = ',endwb_grc(indexg)-begwb_grc(indexg)
+             write(iulog,*)'deltawb/dtime             = ',(endwb_grc(indexg)-begwb_grc(indexg))/dtime
+             write(iulog,*)'qflx_drain_perched        = ',qflx_drain_perched_grc(indexg)*dtime
+             write(iulog,*)'forc_flood                = ',forc_flood_grc(indexg)*dtime
+             write(iulog,*)'qflx_glcice_dyn_water_flux = ',qflx_glcice_dyn_water_flux_grc(indexg)*dtime
+
+             write(iulog,*)'CTSM model is stopping'
+             call endrun(decomp_index=indexg, clmlevel=nameg, msg=errmsg(sourcefile, __LINE__))
+          end if
+
+       end if
+
+       ! Snow balance check at the column level.
+       ! Beginning snow balance variable h2osno_old is calculated once
+       ! for both the column-level and grid cell-level balance checks.
```

Comment:
Removing the 2nd and 3rd line of this comment. For some reason I had convinced myself that there was a grc-level balance check for snow...